kernel/src/features/mod.rs

```diff
@@ -31,6 +31,11 @@ pub enum ReaderFeatures {
     #[strum(serialize = "timestampNtz")]
     #[serde(rename = "timestampNtz")]
     TimestampWithoutTimezone,
+    // Allow columns to change type
+    TypeWidening,
```

Comment:
It looks like we "trivially" support type widening table feature, by automatically widening any type that needs it regardless of whether the table actually enables the feature? I guess the eventual write path would need to be more strict, but that doesn't exist yet so it's a problem for a different PR.

Reply:
Yes, there's no point in crippling upcasting on read if type widening isn't enabled. The feature mostly ensures that we have consistent support by specifying what type changes must be supported.